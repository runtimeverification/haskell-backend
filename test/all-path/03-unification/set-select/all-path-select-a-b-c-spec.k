requires "path.k"

module ALL-PATH-SELECT-A-B-C-SPEC
  imports PATH

  // This should be provable as an one-path claim, but should not be provable as an
  //all-path claim.
  claim <k> select => b #Or c </k>
       <state> SetItem(a) SetItem(b) SetItem(c) </state> [all-path]

endmodule
