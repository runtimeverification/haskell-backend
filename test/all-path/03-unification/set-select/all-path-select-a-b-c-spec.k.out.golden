<generatedTop>
  <k>
    a ~> .K
  </k>
  <state>
    SetItem ( a )
    SetItem ( b )
    SetItem ( c )
  </state>
</generatedTop>
